SSA/Projects/InstCombine/scripts/mlir-tool.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.dialects.llvm import LLVM
+from xdsl.utils.exceptions import ParseError
+from xdsl.context import MLContext
+from xdsl.dialects import get_all_dialects
+from xdsl.dialects.llvm import FuncOp
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.dialects.builtin import (
+    Builtin,
+    IndexType,
+    IntegerAttr,
+    IntegerType,
+    ModuleOp,
+    StringAttr,
+    i32,
+    i64,
+)
+import os
+import io
+from pathlib import Path
+from xdsl.printer import Printer
+
+# Initialize the MLIR context and register the LLVM dialect
+ctx = MLContext(allow_unregistered=True)
+ctx.load_dialect(LLVM)
+ctx.load_dialect(Builtin)
+
+directory = os.fsencode("../vcombined-mlir")
+
+
+allowed_names = {
+    "llvm.return",
+    "llvm.mul",
+    "llvm.add",
+    "llvm.sub",
+    "llvm.shl",
+    "llvm.and",
+    "llvm.or",
+    "llvm.xor",
+    "llvm.mlir.constant",
+    "llvm.lshr",
+    "llvm.ashr",
+    "llvm.urem",
+    "llvm.srem",
+    "llvm.add",
+    "llvm.mul",
+    "llvm.sub",
+    "llvm.sdiv",
+}
+allowed_unregistered = set()  # {
+
+
+# "llvm.icmp"
+# }
+def allowed(op):
+    return (
+        hasattr(op, "sym_name")
+        or (
+            op.name == "builtin.unregistered"
+            and op.op_name.data in allowed_unregistered
+        )
+        or (op.name in allowed_names)
+    )
+
+
+def show(block):
+    output = io.StringIO()
+    p = Printer(stream=output)
+    block.print(p)
+    contents = output.getvalue()
+    output.close()
+    return contents
+
+
+def showr(region):
+    output = io.StringIO()
+    p = Printer(stream=output)
+    p.print_region(region)
+    contents = output.getvalue()
+    output.close()
+    return contents
+
+
+def size(func):
+    s = 0
+    for _ in func.walk():
+        s += 1
+    return s
+
+
+for file in os.listdir(directory):
+    filename = os.fsdecode(file)
+    print(filename)
+    stem = "g" + filename.split(".")[0].replace("-", "h")
+    output = ""
+    with open(os.path.join("../vcombined-mlir", filename), "r") as f1:
+        with open(os.path.join("../vbefore-mlir", filename), "r") as f2:
+            data = f1.read()
```

Comment:
seems like it would be good to be consistent with bla1 and bla2 everywhere
```suggestion
            data1 = f1.read()
```